src/sentry/static/sentry/app/components/onboardingWizard/task.tsx

```diff
@@ -0,0 +1,287 @@
+        {task.title}
+      </Title>
+      <Description>{`${task.description}. ${task.detailedDescription}`}</Description>
+      {task.requisiteTasks.length === 0 && (
+        <ActionBar>
+          {task.status === 'pending' ? (
+            <InProgressIndicator user={task.user} />
+          ) : (
+            <CTA>Setup now</CTA>
+          )}
+          {skipAction}
+          {supplement}
+        </ActionBar>
+      )}
+    </Item>
+  );
+}
+
+const Item = styled(Card)`
+  padding: ${space(3)};
+  position: relative;
+`;
+
+const Title = styled('h5')`
+  font-weight: normal;
+  display: grid;
+  grid-template-columns: max-content 1fr;
+  grid-gap: ${space(0.75)};
+  align-items: center;
+  margin: 0;
+`;
+
+const Description = styled('p')`
+  padding-top: ${space(1)};
+  font-size: ${p => p.theme.fontSizeSmall};
+  line-height: 1.75rem;
+  color: ${p => p.theme.gray3};
+  margin: 0;
+`;
+
+const ActionBar = styled('div')`
+  height: 40px;
+  border-top: 1px solid ${p => p.theme.borderLight};
+  margin: ${space(3)} -${space(3)} -${space(3)};
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 0 ${space(2)};
+`;
+
+const InProgressIndicator = styled(
+  ({user, ...p}: React.HTMLAttributes<HTMLDivElement> & {user?: AvatarUser | null}) => (
+    <div {...p}>
+      <Tooltip
+        disabled={!user}
+        containerDisplayMode="flex"
+        title={tct('This task has been started by [user]', {
+          user: user?.name,
+        })}
+      >
+        <IconEvent />
+      </Tooltip>
+      {t('Task in progress...')}
+    </div>
+  )
+)`
+  font-size: ${p => p.theme.fontSizeMedium};
+  font-weight: bold;
+  color: ${p => p.theme.yellowOrange};
+  display: grid;
+  grid-template-columns: max-content max-content;
+  align-items: center;
+  grid-gap: ${space(1)};
+`;
+
+const CTA = styled('div')`
+  color: ${p => p.theme.blue};
+  font-size: ${p => p.theme.fontSizeMedium};
+  font-weight: bold;
+`;
+
+const SkipButton = styled(Button)`
+  color: ${p => p.theme.gray2};
+`;
+
+const TaskLockedIndicator = styled(IconLock)``;
+
+TaskLockedIndicator.defaultProps = {
+  size: 'xs',
+  color: theme.redLight,
+};
+
+const PosedItemComplete = posed(Card)({
+  complete: {staggerChildren: 500},
+});
+
+const ItemComplete = styled(PosedItemComplete)`
+  cursor: pointer;
+  color: ${p => p.theme.gray3};
+  padding: ${space(1)} ${space(1.5)};
+  display: grid;
+  grid-template-columns: max-content 1fr max-content 20px;
+  grid-gap: ${space(1)};
+  align-items: center;
+`;
+
+const completedItemPoses = {
+  completeInit: {
+    opacity: 0,
+    x: -10,
+  },
+  complete: {
+    opacity: 1,
+    x: 0,
+  },
+};
+
+const CompleteIndicator = posed(IconCheckmark)(completedItemPoses);
+CompleteIndicator.defaultProps = {
+  circle: true,
+  color: theme.green,
+};
+
+const SkippedIndicator = posed(IconClose)(completedItemPoses);
+SkippedIndicator.defaultProps = {
+  circle: true,
+  color: theme.yellowOrange,
+};
```

Comment:
I can't decide if this makes more sense than just passing the props during render or not

Reply:
🤷‍♂ I did this to keep the component render invocation short, since `SkippedIndicator` semantically describes something that doesn't need to have props.